service/src/main/java/com/commercetools/pspadapter/payone/domain/payone/PayonePostServiceImpl.java

```diff
@@ -58,6 +68,43 @@ public static PayonePostServiceImpl of(final String payoneServerApiUrl) throws I
         }
     }
 
+    /**
+     * Expand entries with list values to key-value pairs where the keys are transformed to set of {@code key[i]} with
+     * respective values from the list. Non-list arguments remain the same.
+     * <p>
+     * The indices origin is <b>1</b>.
+     * <p>
+     * <b>Note:</b> so far only {@link List} values are expanded, arrays behavior is still undefined.
+     * <p>
+     * Example:
+     * <pre> {"a": [5, 7], "b": "c", "d": ["e", "f"]} =>
+     * {"a[1]": 5, "a[2]": 7, "b": "c", "d[1]": "e", "d[2]": "f"}
+     * </pre>
+     *
+     * @param parameters Set of key-value pairs to expand
+     * @return new {@link Map} where keys with list values
+     */
+    @Nonnull
+    Map<String, Object> getObjectMapWithExpandedLists(Map<String, Object> parameters) {
```

Comment:
What speaks against putting data directly into map as `map.add("a[1]", 5);` instead of converting it here again?